spec/features/users_feature_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+context 'user not signed in and on the homepage' do
+  it 'should see a "sign in" link and a "sign up" link' do
+    visit('/')
+    expect(page).to have_link('Sign in')
+    expect(page).to have_link('Sign up')
+  end
+
+  it 'should not see "sign out" link' do
+    visit('/')
+    expect(page).not_to have_link('Sign out')
+  end
+
+  it 'should not be able to create a new post' do
+    visit('/')
+    click_link('Post to Instagram')
+    expect(page).to have_content('You need to sign in or sign up before continuing')
```

Comment:
Metrics/LineLength: Line is too long. [84/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)